packages/vscode-extension/resources/scripts/connectionPanel.js

```diff
@@ -0,0 +1,38 @@
+(() => {
+  document.getElementById('test-connection').addEventListener('click', () => {
+    vscode.postMessage({
+      command: 'onTestConnection',
+      connection: this.getConnection(),
+      password: this.getPassword(),
+    });
+  });
+
+  document.getElementById('add-connection').addEventListener('click', () => {
+    vscode.postMessage({
+      command: 'onAddConnection',
+      connection: this.getConnection(),
+      password: this.getPassword(),
+    });
+  });
+
+  getConnection = () => {
+    const name = document.getElementById('name').value;
+    let key = document.getElementById('key').value;
+    key = !key ? name.toLowerCase().replace(/\s/g, '-') : key;
```

Comment:
keys must remain unique through name changes.. this is flawed.. it needs some unique slug to ensure that a connection with the same name doesn't get the same key

Reply:
Why not generate a unique uuid instead?